### PR TITLE
Some assembly requaired

### PR DESCRIPTION
**Pull request by Razharas**
Changes wirecutters for an igniter in a stunprod
Stunprod sparkles on proding
Timer now have "repeating" setting, will restart again and again after
expiring
Voice analyzer got exclusive mode that works like a password, i.,e. need
a perfect match of the phrase
Voice analyzer got voice recognition mode that remembers human voice
instead of speech, works like personality recognition
Voice analyzer got a sensor mode, works on any speech it hears from
anyone
Voice analyzer modes and changed by applying a multitool to it

### CODE REVIEW

**Comment by Iamgoofball** [comment]
can the voice analyzer pick up voice changer fakes as the real deal with voice recognition mode?

**Comment by Lumipharon** [comment]
The stun prod change makes literally no sense.

**Comment by Razharas** [comment]
Nope, it just saved mob and checks it
Suggest alternatives

**Comment by Buggy123** [comment]
Nice but stunprod change is unnecessary and bizarre. In what way does a stun prod use a IGNITER.

**Comment by Iamgoofball** [comment]
have the voice analyzer check the actual name supplied, I'm pretty sure Hear() has some shit where it passes that

paging @miauw62 the saycode expert

**Comment by DaedalusGame** [comment]
le igniter producing sparks meme

**Comment by MMMiracles** [comment]
>requaired

**Comment by Buggy123** [comment]
Yeah I guess a igniter makes like a burst of high voltage or something and thats why it sparks etc but its dumb, wirecutters make perfect sense.

**Comment by Razharas** [comment]
Im just trying to push some usability upon assemblies guise dont appeal to realism plz

**Comment by Razharas** [comment]
The name match sounds interesting

**Comment by PKPenguin321** [comment]
:-1: to the stunprod change, it's dumb and makes no sense and seems to be change for the sake of change
other stuff is cool though

**Comment by KazeEspada** [comment]
Why not both igniters and wirecutters!

**Comment by PKPenguin321** [comment]
>So this changes nothing.

not necessarily, i often have a pair of wirecutters in my toolbelt for making prods on the fly if the need arises, an igniter not so much
besides, it's really just a change for the sake of change. there's really no point in doing it

**Comment by Lumipharon** [comment]
why does an igniter have anything to do with how a prod works?

**Comment by duncathan** [comment]
Igniters use sparks to ignite so naturally they would have two conducive ends through which a current can flow; just like wirecutters do. 
The point of the change is to make usage of assemblies more prevalent. 

Seems pretty obvious to me.

**Comment by PKPenguin321** [comment]
i dont get why its necessary to change to igniters though
like why

**Comment by duncathan** [comment]
>The point of the change is to make usage of assemblies more prevalent.

**Comment by Lumipharon** [comment]
how does it make the use of assemblies more prevalent?

An igniter is not an assembly. It's an igniter, which logically would have far smaller components with which you could actually zap someone with, as opose to a big ass set of wire cutters.

**Comment by MMMiracles** [comment]
It just occurred to me that you never bother to strip off the insulation on the grip of the wirecutters when building a stunprod.

please fix.

**Comment by Lumipharon** [comment]
they're not insulated though. Try snip snip some wires without gloves.

Just like wires, NT probably just painted them to look insulated.

**Comment by DaedalusGame** [comment]
they're insulated in highly conductive polymerized plasma.

File: code/modules/assembly/voice.dm [hunk 1]
```diff
@@ -11,20 +11,65 @@
 	verb_exclaim = "beeps"
 	var/listening = 0
 	var/recorded = "" //the activation message
+	var/mode = 1
+	var/global/list/modes = list("inclusive",
```
Comment (neersighted) [inline code/modules/assembly/voice.dm]: defines pls

**Comment by Miauw62** [comment]
WHO SUMMONS ME

File: code/modules/assembly/voice.dm (hunk 1 above)
Comment (Miauw62) [inline code/modules/assembly/voice.dm]: this ^^^^^ ^^^ ^6 ^^

File: code/modules/assembly/voice.dm [hunk 2]
```diff
@@ -11,20 +11,65 @@
...
+		record_speech(speaker, raw_message)
 	else
-		if(findtextEx(raw_message, recorded))
+		if(check_activation(speaker, raw_message))
 			spawn(10)
 				pulse(0)
 
+/obj/item/device/assembly/voice/proc/record_speech(atom/movable/speaker, raw_message)
+	switch(mode)
+		if(1)
```
Comment (Miauw62) [inline code/modules/assembly/voice.dm]: ew

File: code/modules/assembly/voice.dm [hunk 3]
```diff
@@ -11,20 +11,65 @@
...
+	. = 0
+	switch(mode)
+		if(1)
+			if(findtextEx(raw_message, recorded))
+				. = 1
+		if(2)
+			if(raw_message == recorded)
+				. = 1
+		if(3)
+			if(speaker == recorded)
```
Comment (Miauw62) [inline code/modules/assembly/voice.dm]: this doesnt work with radios, pls fix

File: code/game/objects/items/weapons/weaponry.dm [hunk 4]
```diff
@@ -117,7 +117,7 @@
 		qdel(I)
 		qdel(src)
 
-	else if(istype(I, /obj/item/weapon/wirecutters) && !(I.flags & NODROP))
+	else if(istype(I, /obj/item/device/assembly/igniter) && !(I.flags & NODROP))
```
Comment (neersighted) [inline code/game/objects/items/weapons/weaponry.dm]: Remove nodrop check.

File: code/game/objects/items/weapons/weaponry.dm [hunk 5]
```diff
@@ -125,7 +125,7 @@
 		user.unEquip(I)
```
Comment (neersighted) [inline code/game/objects/items/weapons/weaponry.dm]:
```
if(!user.drop_item())
    return
```
Checks nodrop for you

File: code/game/objects/items/weapons/weaponry.dm [hunk 6]
```diff
@@ -117,7 +117,7 @@
 		qdel(I)
 		qdel(src)
 
-	else if(istype(I, /obj/item/weapon/wirecutters) && !(I.flags & NODROP))
+	else if(istype(I, /obj/item/device/assembly/igniter) && !(I.flags & NODROP))
 		var/obj/item/weapon/melee/baton/cattleprod/P = new /obj/item/weapon/melee/baton/cattleprod
 
 		if(!remove_item_from_storage(user))
```
Comment (neersighted) [inline code/game/objects/items/weapons/weaponry.dm]: We don't use this anymore.

**Comment by Razharas** [comment]
I can buff batons in some way if you think using igniter is a nerf

**Comment by Lumipharon** [comment]
It's not a nerf, it's just a pointless change.

It's like if you changed flamethrowers to use a wrench instead of a welder. It doesn't make any sense and has no balance effect. It's change for the sake of change.

**Comment by PKPenguin321** [comment]
^

**Comment by Razharas** [comment]
Its a change to make igniters have some uses outside stuff you can use only as antag

**Comment by PKPenguin321** [comment]
why bother though, we have plenty of things with antag-only uses

**Comment by PKPenguin321** [comment]
I CHANGED MY MIND THIS IS SOME GOOD SHIT
:+1:👀:+1:👀:+1:👀:+1:👀:+1:👀 good shit go౦ԁ sHit:+1: thats ✔ some good:+1::+1:shit right:+1::+1:there:+1::+1::+1: right✔there ✔✔if i do ƽaү so my self 💯 i say so 💯 thats what im talking about right there right there (chorus: ʳᶦᵍʰᵗ ᵗʰᵉʳᵉ) mMMMMᎷМ💯 :+1::+1: :+1:НO0ОଠOOOOOОଠଠOoooᵒᵒᵒᵒᵒᵒᵒᵒᵒ:+1: :+1::+1: :+1: 💯 :+1: 👀 👀 👀 :+1::+1:Good shit

**Comment by Lumipharon** [comment]
>Its a change to make igniters have some uses outside stuff you can use only as antag
>flamethrowers can never be used by non antags
>grenades can never be used by non antags
>ore redemption machines are never built by non antags

And all that aside, is having an item that isn't commonly used by non antags even an issue? Like really? Lets make tyrant and antimov boards glow so non antags can use them as flashlights :^)

**Comment by duncathan** [comment]
Clearly no amount of obvious reasoning will get to you two

**Comment by phil235** [comment]
no changelog?

**Comment by Razharas** [comment]
Dunc merged it so fast i neither added all the things i wanted nor been able to make changelog
I can add it in the second pr

**Comment by duncathan** [comment]
My bad
I'm gonna take a break from merging things for a day or two I need to cool off

**Comment by Lumipharon** [comment]
Still no one has explained an actual reason for the prod changes other than "I want a completely unrelated item to have more uses".

**Pull request merged**